beyond-all-reason/Beyond-All-Reason
# Perf!: Optimize commander select widget

### Work done
Instead of relying on unitadded and unitremoved callins - to manage internal state - incurring in performance overhead during idle game runtime, we do all the work inside the action handler itself.

Also, allow users to customize the behavior they want with the following action args:

- focus: whether to focus camera on selected unit
- includeselected: whether selected units are part of the selection pool
- append: whether to append unit to current selection, instead of replacing

#### Test setup

- `/cheat`
- `/give corcom`
- `/give armcom`
- Use grid keys

#### Test steps

- Press tab, confirm that it cycles between the three commanders
- Select two commanders, press tab, confirm that it selects the only unselected commander
- Select three commanders, press tab, confirm nothing happens
- Select no commanders, hold shift, confirm pressing tab three times selects all commanders
